File: 88-merge-sorted-array/merge-sorted-array.cpp
class Solution {
public:
    void merge(vector<int>& nums1, int m, vector<int>& nums2, int n) {
        int last_index = m + n - 1;

        while(m>0 && n>0){
            if (nums1[m-1] >= nums2[n-1]){
                nums1[last_index] = nums1[m-1];
                last_index--;
                m--;
            }
            else{
                nums1[last_index] = nums2[n-1];
                last_index--;
                n--;
            }
        }

        while (n>0){
            nums1[last_index] = nums2[n-1];
            last_index--;
            n--;
        }


    }
};

File: 88-merge-sorted-array/merge-sorted-array.py
class Solution:
    def merge(self, nums1: List[int], m: int, nums2: List[int], n: int) -> None:
        """
        Do not return anything, modify nums1 in-place instead.
        """
        p1 = 0
        p2 = 0
        temp = []

        while (p1<m and p2<n):
            if nums1[p1]<nums2[p2]:
                temp.append(nums1[p1])
                p1+=1

            else:
                temp.append(nums2[p2])
                p2+=1

        while p1<m:
            temp.append(nums1[p1])
            p1+=1

        while p2<n:
            temp.append(nums2[p2])
            p2+=1

        for i in range(0, m+n):
            nums1[i] = temp[i]


        